scipy/integrate/_ode.py

```diff
@@ -802,24 +801,24 @@ def _determine_mf_and_set_bands(self, has_jac):
     def reset(self, n, has_jac):
         mf = self._determine_mf_and_set_bands(has_jac)
 
-        if mf == 10:
+        if type(mf) != int or mf < 10 or mf > 25 or (mf >= 16 and mf <= 19):
+            raise ValueError('Unexpected mf=%s' % mf)
+        elif mf == 10:
             lrw = 20 + 16 * n
-        elif mf in [11, 12]:
+        elif mf <= 12:
             lrw = 22 + 16 * n + 2 * n * n
         elif mf == 13:
             lrw = 22 + 17 * n
-        elif mf in [14, 15]:
+        elif mf <= 15:
             lrw = 22 + 18 * n + (3 * self.ml + 2 * self.mu) * n
         elif mf == 20:
             lrw = 20 + 9 * n
-        elif mf in [21, 22]:
+        elif mf <= 22:
             lrw = 22 + 9 * n + 2 * n * n
         elif mf == 23:
             lrw = 22 + 10 * n
-        elif mf in [24, 25]:
-            lrw = 22 + 11 * n + (3 * self.ml + 2 * self.mu) * n
         else:
-            raise ValueError('Unexpected mf=%s' % mf)
+            lrw = 22 + 11 * n + (3 * self.ml + 2 * self.mu) * n
```

Comment:
I prefer the previous construction. It may be slightly less efficient, but it's much easier to understand what each branch is doing without reading the whole thing. Also checking `type(mf)` might break existing code.